doc: openthread: Adding documentation about TCAT. (#14254)
nrfconnect/sdk-nrf: doc/nrf/protocols/thread/tcat.rst

.. _thread_tcat:

Thread Commissioning Over Authenticated TLS (TCAT)
##################################################

Developed to address the needs of professional installation and commercial building scenarios,
commissioning at scale makes it easy to wirelessly onboard Thread devices that are pre-installed in difficult-to-reach places, such as inside a ceiling or embedded in a wall.
Instead of scanning physical install codes once installed devices are powered,
Thread commissioning can be implemented over authenticated TLS (Transport Layer Security) by exchanging security certificates.

> This authenticated commissioning can be performed while in proximity, using a mobile device (phone, tablet) over a wireless connection such as a Bluetooth Low Energy radio.

Author: fixed.

Building and Running
********************

Device site

> ===========

Author: fixed.

> Currently NCS provides experimental support for TCAT. Compile CLI sample with TCAT snippet enabled: :ref:`ot_cli_sample_activating_variants`

Reviewer: ```suggestion
The |NCS| currently provides experimental support for TCAT.
To test this feature, build the :ref:`ot_cli_sample` sample with the :ref:`TCAT snippet <ot_cli_sample_activating_variants>` enabled.
```

> After flashing the device this feature can be anabled in following way using CLI.
> 
>     .. code-block:: console
>         uart:~$ ot tcat start
> 

Reviewer: ```suggestion
After flashing the sample to the device, use the following command to enable TCAT:

.. code-block:: console
   uart:~$ ot tcat start

```

Author: fixed.

> Commissioner site
> =================

Reviewer: ```suggestion
```

Author: fixed.

> Currently the only available commissioner tool can be found in OpenThread repository. Please refer to instruction on how to use it:
> https://github.com/nrfconnect/sdk-openthread/tree/main/tools/tcat_ble_client

Reviewer: ```suggestion
Currently, `BBTC Client`_ is the only available TCAT commissioner tool, and can be found in the OpenThread repository.
Refer to the tool's documentation for more information.
```
Then the following needs to be added to Line 100 of `links.txt`:
```
.. _`BBTC Client`: https://github.com/nrfconnect/sdk-openthread/tree/main/tools/tcat_ble_client
```

Author: fixed.
